src/acuvity/guard/config.py

```diff
@@ -0,0 +1,279 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+
+from .constants import GuardName
+from .errors import ConfigValidationError, GuardConfigError, ThresholdParsingError
+from .threshold import Threshold
+
+
+@dataclass(frozen=True)
+class Match:
+    """Immutable match configuration"""
+    threshold: Threshold
+    redact: bool = False
+    count_threshold: int = 0
+
+@dataclass(frozen=True)
+class Guard:
+    """Immutable guard configuration"""
+    name: GuardName
+    matches: Dict[str, Match]
+    threshold: Threshold
+    count_threshold: int = 0
+
+class GuardConfig:
+    """
+    Parser for guard configuration files.
+
+    This class handles parsing of guard configuration files in YAML format,
+    validating their contents, and converting between analyzer names and IDs.
+
+    The parser handles two types of guards:
+    1. Match Guards: Guards with a 'matches' section (e.g., pii_detector)
+    2. Simple Guards: Guards without matches (e.g., prompt_injection, toxicity)
+    """
+
+    DEFAULT_THRESHOLD = Threshold("> 0.0")
```

Comment:
I thought we said we want to include everything? shouldn't it be `>= 0.0` then?

Reply:
yes.